ui-manchette/src/components/Manchette.tsx

```diff
@@ -26,40 +29,82 @@ const Manchette = ({
   resetZoom,
   yZoom = 1,
   waypoints,
-  activeWaypointId,
+  waypointMenuData,
   isProportional = true,
   toggleMode,
   children,
   height = INITIAL_OP_LIST_HEIGHT,
-}: ManchetteProps) => (
-  <div className="manchette-container">
-    <div className="bg-white-100 border-r border-grey-30" style={{ height: `${height}px` }}>
-      <WaypointList waypoints={waypoints} activeWaypointId={activeWaypointId} />
-      {children}
-    </div>
-    <div className="manchette-actions">
-      <div className="zoom-buttons">
-        <button className="zoom-out" onClick={zoomYOut} disabled={yZoom <= MIN_ZOOM_Y}>
-          <ZoomOut />
-        </button>
-        <button className="zoom-in" onClick={zoomYIn} disabled={yZoom >= MAX_ZOOM_Y}>
-          <ZoomIn />
-        </button>
-        <button className="zoom-reset" onClick={resetZoom}>
-          Fit
-        </button>
+}: ManchetteProps) => {
+  const [menuPosition, setMenuPosition] = useState<number>();
+
+  // Allow to track the menu position after we might have scrolled in the page
+  useEffect(() => {
```

Comment:
We measure DOM elements from inside this function, so we should use `useLayoutEffect()` instead, so that our callback isn't called before the DOM is updated.

Reply:
Still got to keep this one in mind, thank you